scripts/build-doc-list.js

```diff
@@ -114,3 +129,28 @@ function getReleaseDate(text) {
  const releaseDate = `${splittedText[1]}-${splittedText[2]}` // '2021-06'
  return releaseDate
 }
+
+function checkAndGroupDocumentsByCategory(results) {
+  return results.reduce((items, item, index) => {
+
+    const { menu: { docs: { parent: documentCategoryName } = {} } = {} } = item;
+
+    if (documentCategoryName) {
+      const documentCategory = result.find((item) => item.title === documentCategoryName);
+      const categoryHasBeenProcessed = items.find((item) => item.title === documentCategoryName);
+      if (!documentCategory) {
+        if(!categoryHasBeenProcessed) {
```

Comment:
You have nested single if :)

```suggestion
      if (!documentCategory && !categoryHasBeenProcessed) {
```

Reply:
Cool thanks @magicmatatjahu I have changed this stuff slightly and refactored out into new files, as I wasnt really happy with the solution, hopefully its cleaner now